frontend/src/components/SystemsTable/SystemFilter.tsx

```diff
@@ -0,0 +1,181 @@
+const filterDelim = "..";
+export const filterKeyMap = {
+  nameFilter: "name",
+  taskFilter: "task",
+  showMine: "show_mine",
+  sortField: "sort_field",
+  sortDir: "sort_dir",
+  dataset: "dataset",
+  datasetSplit: "dataset_split",
+  subdataset: "sub_dataset",
+  activeSystemIDs: "system_id",
+};
+
+export type FilterUpdate = Partial<SystemFilter>;
+export type QueryDict = { [key: string]: string };
+
+export class SystemFilter {
+  name: string;
+  task: string;
+  showMine: boolean;
+  sortField: string;
+  sortDir: "asc" | "desc";
+  dataset: string;
+  split: string;
+  subdataset: string;
+  activeSystemIDs: Array<string>;
+
+  constructor(partial: Partial<FilterUpdate> | null) {
+    if (!partial) {
+      this.name = "";
+      this.task = "";
+      this.showMine = false;
+      this.sortField = "created_at";
+      this.sortDir = "desc";
+      this.dataset = "";
+      this.split = "";
+      this.subdataset = "";
+      this.activeSystemIDs = [];
+      return;
+    }
+
+    const defaultFilter = SystemFilter.getDefaultFilter();
+    this.name = partial.name || defaultFilter.name;
+    this.task = partial.task || defaultFilter.task;
+    this.showMine =
+      partial.showMine === undefined
+        ? defaultFilter.showMine
+        : partial.showMine;
+    this.sortField = partial.sortField || defaultFilter.sortField;
+    this.sortDir = partial.sortDir || defaultFilter.sortDir;
+    this.dataset = partial.dataset || defaultFilter.dataset;
+    this.split = partial.split || defaultFilter.split;
+    this.subdataset = partial.subdataset || defaultFilter.subdataset;
+    this.activeSystemIDs =
+      partial.activeSystemIDs || defaultFilter.activeSystemIDs;
+  }
+
+  /*  Update the filter with the values in the new partial filter 
+      Returns true if >= 1 of current filter values get updated
+
+      TODO: Rolling out all if statement seems necessary because Typescript
+      won't allow variable field access for classes. Is there a bette way tho?
+  */
+  update(partial: FilterUpdate): boolean {
```

Comment:
You can use `Object.getOwnPropertyNames` to get all the property names. All the regular stuff still works on class objects. In this case, I think this method can be written this way:
```
update(newFilters: FilterUpdate) {
  const filterValues = {...this, ...newFilters};
  return new SystemFiler(filterValues);
}
``` 
1. I think it'll make the code simpler if we return a new `SystemFilter` object because we store this object as a state and we need to create a new object whenever the state changes.
2. Checking if there is actually an update may not be necessary because we are passing in the values that needs to be updated only. Looks like this check is not very straightforward (you need to `JSON.stringfy` and compare some of the fields) so l think we can maybe skip it.